etna/models/nn/utils.py

```diff
@@ -236,7 +236,33 @@ def fit(self, ts: TSDataset):
             raise ValueError("Trainer or model is None")
         return self
 
+    def _get_first_prediction_timestamp(self, ts: TSDataset, horizon: int) -> pd.Timestamp:
+        return ts.index[-horizon]
+
+    def _is_in_sample_prediction(self, ts: TSDataset, horizon: int) -> bool:
+        first_prediction_timestamp = self._get_first_prediction_timestamp(ts=ts, horizon=horizon)
+        return first_prediction_timestamp <= self._last_train_timestamp
+
+    def _is_prediction_with_gap(self, ts: TSDataset, horizon: int) -> bool:
+        first_prediction_timestamp = self._get_first_prediction_timestamp(ts=ts, horizon=horizon)
+        first_timestamp_after_train = pd.date_range(self._last_train_timestamp, periods=2, freq=self._freq)[-1]
```

Comment:
May be we can solve it with timedelta?

Reply:
I don't think it is a good idea. Frequency works more subtle than simple timedelta, as I understand.